tests/test_fixtures.py

```diff
@@ -0,0 +1,769 @@
+            request_handler.send_response(200)
+            request_handler.end_headers()
+            request_handler.wfile.write(b'malformed image')
+
+            self._failure_counter += 1
+        else:
+            request_handler.default_get()
+
+    def map(self, url):
+        return {'GET': {url if url else self._url: MalformedImageHandler(self._number_of_failures)}}
+
+
+class SlowRetrievalHandler:
+    def __init__(self, number_of_failures=1, url=''):
+        self._number_of_failures = number_of_failures
+        self._failure_counter = 0
+        self._url = url
+
+    def __call__(self, request_handler):
+        if self._failure_counter < self._number_of_failures:
+            request_handler.send_response(200)
+            file_size = os.path.getsize(request_handler.file_path)
+            request_handler.end_headers()
+
+            with open(request_handler.file_path, 'rb') as source:
+                while True:
+                    data = source.read(1)
+                    if not data:
+                        break
+                    request_handler.wfile.write(data)
+                    request_handler.wfile.flush()
+                    import time
+                    time.sleep(100)
+
+            self._failure_counter += 1
+        else:
+            request_handler.default_get()
+
+    def map(self, url):
+        return {'GET': {url if url else self._url: SlowRetrievalHandler(self._number_of_failures)}}
+
+
+class RedirectHandler:
+    def __init__(self, number_of_redirects=1, url=''):
+        self._number_of_redirects = number_of_redirects
+        self._redirect_counter = 0
+        self._url = url
+
+    def __call__(self, request_handler):
+        if self._redirect_counter < self._number_of_redirects:
+            request_handler.send_response(301)
+            request_handler.send_header('Location', request_handler.server.base_url + request_handler.path)
+            request_handler.end_headers()
+            self._redirect_counter += 1
+        else:
+            request_handler.default_get()
+
+    def map(self, url):
+        return {'GET': {url if url else self._url: RedirectHandler(self._number_of_redirects)}}
+
+
+class MalformedJsonHandler:
+    def __init__(self, number_of_failures=1):
+        self._number_of_failures = number_of_failures
+        self._failure_counter = 0
+
+    def __call__(self, request_handler):
+        if self._failure_counter < self._number_of_failures:
+            request_handler.send_response(200)
+            request_handler.end_headers()
+            request_handler.wfile.write(b'{"non-uptane-json": "some-value"}')
+
+            self._failure_counter += 1
+        else:
+            request_handler.default_get()
+
+    def map(self, url):
+        return {'GET': {url: MalformedJsonHandler(self._number_of_failures)}}
+
+
+# The following are candidates for fixtures/common/shared components that can be reused by
+# multiple tests/test suits
```

Comment:
Aren't they already used by multiple tests now? :)

Reply:
Yes, this comment was introduced back then when I produced initial tests for IP secondary, I'll remove it.